From 76ea096ff56ac38f9a5b90bdd37d0cfdfe60127c Mon Sep 17 00:00:00 2001
From: Paritosh <paritosh.debnath@intel.com>
Date: Thu, 19 Sep 2013 15:50:13 +0530
Subject: Fixing memory leak when DEBUG EVENT is received

Certain commands give Dubug Event which are ignored in
current implementation. This fix will release the memory
before ignoring it.

Change-Id: I16ff5c0a99450671f8fec0986c66e41b4744ffd0
Signed-off-by: Paritosh <paritosh.debnath@intel.com>
---
 hci/src/hci_h4.c |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/hci/src/hci_h4.c b/hci/src/hci_h4.c
index cf141b5..d1cba86 100644
--- a/hci/src/hci_h4.c
+++ b/hci/src/hci_h4.c
@@ -343,7 +343,12 @@ uint8_t internal_event_intercept(void)
 
     else if (event_code == HCI_INTEL_DEBUG_EVT)
     {
-        //ignore the event
+        //Ignore the event and release p_rcv_msg buffer
+        if (bt_hc_cbacks)
+        {
+            bt_hc_cbacks->dealloc((TRANSAC) p_cb->p_rcv_msg, \
+                (char *) (p_cb->p_rcv_msg + 1));
+        }
         return IGNORE_EVENT;
     }
 
-- 
1.7.4.1

